### PR TITLE
Allows an absolute URL for the authorizationPath

### PR DESCRIPTION
**Pull request by gillesdemey**
Partially adresses #55, #29 and #40.

Switched from `url.resolve` to a different method, since invocation of `url.resolve('http://foo.bar/baz/', 'qux')` would result in `http://foo.bar/qux`.

### CODE REVIEW

File: lib/client/auth-code.js [hunk 1]
```diff
@@ -18,7 +19,13 @@ module.exports = function (config) {
     params.response_type = 'code';
     params.client_id = config.clientID;
 
-    return config.site + config.authorizationPath + '?' + qs.stringify(params);
+    var isAbsolute = String(Url.parse(config.authorizationPath).protocol)
```
Comment (jonathansamines) [inline lib/client/auth-code.js]: I thought that using `url.resolve` to allow this was easier?

Comment (gillesdemey) [inline lib/client/auth-code.js]: 'url.resolve' will resolve a url like a browser would, but if the 'site' were to be configured as 'http://domain.com/oauth' and the 'authorizationPath' were '/authorize' it will drop the '/oauth' from the path.

Apparantly the test suite does not account for this edge-case, hence I didn't see it failing before.

Comment (jonathansamines) [inline lib/client/auth-code.js]: Right now the test suit is only covering high-level cases. I´ll try to improve that in the future. Thanks for this.
